### PR TITLE
tropomi better flags handling

### PR DESCRIPTION
**Pull request by jeromebarre**
Update the tropomi no2 converter to have two new optional arguments to handle the qa_value flaggin and have a thinning capability give the size and the resolution of the dataset.

The new optional arguments are:
`'-q', '--qa_value'` with the documentation link added for the default value in the description as suggested by @srherbener

`'-n', '--thin'` thinning factor with e.g. 0.9 thinning 90% of the data out. Default is 0.0 and takes all obs. 

CI tests results are not affected.

### CODE REVIEW

File: src/compo/tropomi_no2_nc2ioda.py [hunk 1]
```diff
@@ -83,8 +85,10 @@ def _read(self):
             qa_value = qa_value.ravel()
 
             # adding ability to pre filter the data using the qa value
-            # documentation recommends using > 0.75
-            flg = qa_value > 0.75
+            # and also perform thinning using random uniform draw
+            qaf = qa_value > self.qa_flg
```
Comment (fabiolrdiniz) [inline src/compo/tropomi_no2_nc2ioda.py:89]:
```suggestion
            qaf = qa_value > self.qa_flg
```

File: src/compo/tropomi_no2_nc2ioda.py [hunk 2]
```diff
@@ -210,6 +214,19 @@ def main():
         '-c', '--column',
         help="type of column: total or tropo",
         type=str, required=True)
+    optional = parser.add_argument_group(title='optional arguments')
+    optional.add_argument(
+        '-q', '--qa_value',
+        help="qa value used to preflag data the goes into file before QC"
```
Comment (fabiolrdiniz) [inline src/compo/tropomi_no2_nc2ioda.py:220]:
```suggestion
        help="qa value used to preflag data that goes into file before QC"
```

File: src/compo/tropomi_no2_nc2ioda.py [hunk 3]
```diff
@@ -210,6 +214,19 @@ def main():
...
+        "default at 0.75 as suggested in documentation. See:"
```
Comment (fabiolrdiniz) [inline src/compo/tropomi_no2_nc2ioda.py:221]:
```suggestion
        "default at 0.75 as suggested in the documentation. See:"
```

**Review by fabiolrdiniz — APPROVED** [review]
Thanks, @jeromebarre!

**Review by mer-a-o — APPROVED** [review]
Do we want to test these new options in the ctest too?

**Comment by jeromebarre** [comment]
> Do we want to test these new options in the ctest too?

no not needed, it is for running the converter extensively over long period of time. could add them in the ctest in the future to make sure those feature are not breaking over time. i don't have time to do this before the skylab v2 release

**Review by CoryMartin-NOAA — APPROVED** [review]

File: src/compo/tropomi_no2_nc2ioda.py [hunk 4]
```diff
@@ -210,6 +214,19 @@ def main():
...
         help="type of column: total or tropo",
         type=str, required=True)
+    optional = parser.add_argument_group(title='optional arguments')
+    optional.add_argument(
+        '-q', '--qa_value',
+        help="qa value used to preflag data that goes into file before QC"
+        "default at 0.75 as suggested in the documentation. See:"
+        "https://sentinel.esa.int/documents/247904/2474726/"
+        "Sentinel-5P-Level-2-Product-User-Manual-Nitrogen-Dioxide.pdf section 8.6",
+        type=float, default=0.75)
```
Comment (srherbener) [inline src/compo/tropomi_no2_nc2ioda.py:224]: Thanks for adding in the URL!

**Review by srherbener — APPROVED** [review]

**Pull request merged**